src/mlio/default_instance_reader.cxx

```diff
@@ -158,20 +161,58 @@ default_instance_reader::handle_nested_errors()
 std::optional<memory_slice>
 default_instance_reader::read_record_payload()
 {
+    std::string split_record_err_str = "Split record is not formatted with the correct cflag.";
+    if (this->corrupt_split_record_) {
+        throw corrupt_record_error(split_record_err_str); 
+    }
     std::optional<record> rec = read_record();
     if (rec == std::nullopt) {
         return {};
     }
 
-    if (rec->kind() != record_kind::complete) {
-        throw not_supported_error{"Split records are not supported yet!"};
-    }
+    if (rec->kind() == record_kind::complete) {
+        num_bytes_read_ += rec->size();
+        store_record_idx_++;
+
+        return std::move(*rec).payload();
+    } else {
+        if (rec->kind() == record_kind::begin) {
+            std::vector<std::optional<record>> split_records;
+            split_records.push_back(rec);
+            std::size_t total_record_size = 0;
+            total_record_size += rec->size();
+
+            std::optional<record> next_rec = read_record();
+            while(next_rec->kind() == record_kind::middle) {
```

Comment:
Before checking the `kind`, you first need to check whether the record is not `std::nullopt`.

```
while (next_rec && ... ) { ... }
```

Reply:
Changed